python/ray/tune/examples/xgboost_dynamic_resources_example.py

```diff
@@ -22,12 +22,18 @@
 
 def get_best_model_checkpoint(analysis):
     best_bst = xgb.Booster()
-    try:
-        with open(analysis.best_checkpoint, "rb") as inputFile:
-            _, _, raw_model = pickle.load(inputFile)
-        best_bst.load_model(bytearray(raw_model))
-    except IsADirectoryError:
-        best_bst.load_model(os.path.join(analysis.best_checkpoint, CHECKPOINT_FILENAME))
+
+    with analysis.best_checkpoint.as_directory() as checkpoint_dir:
```

Comment:
why does a user have to go through this to get best model checkpoint?
This seems too complicated to showcase as user code.

Reply:
This is for testing with the class trainable cc @Yard1 to maybe adjust this test in the future (e.g. store a separate `model.xgb` in the class trainable) - but likely out of scope for this PR which just adjusts existing logic to the new API.